queries.sql

```diff
@@ -22,4 +22,68 @@ SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with weights that equal precisely 10.4kg or 17.3kg)
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- SECOND PART --
+
+-- Inside a transaction block, update the species of all animals to be 'unspecified', check and then rollback to the initial state --
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Modify animals table: set species to "digimon" for names ending in "mon" and "pokemon" for unspecified species. Confirm changes, commit, and verify persistence. --
+
+BEGIN;
+  UPDATE animals SET species = 'digimon ' WHERE NAME LIKE '%mon';
+  UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+  SELECT * FROM animals ORDER BY id;
+COMMIT;
+  SELECT * FROM animals ORDER BY id;
+
+  -- Inside a transaction delete all records in the animals table, then roll back the transaction. ---
+
+  BEGIN;
+  DELETE FROM animals;
+  SELECT COUNT(id) FROM animals;
+ROLLBACK;
+  SELECT COUNT(id) FROM animals;
+
+ -- Inside a transaction: Remove animals born after Jan 1, 2022. Create savepoint. Change animal weights to negative. Rollback to savepoint. Adjust negative weights. Commit transaction.  --
+
+ BEGIN;
+  DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+  UPDATE animals SET weight_kg = weight_kg * -1;
+  ROLLBACK TO my_savepoint;
+  UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- AGREGATIONS --
+
+-- How many animals are there? --
+SELECT COUNT(id) FROM animals;
+
+
+-- How many animals have never tried to escape? --
+SELECT COUNT(id) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals? --
+SELECT AVG(weight_kg) AS average FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals? --
+SELECT neutered, COUNT(*) AS escape_count FROM animals 
+WHERE escape_attempts > 0 GROUP BY neutered ORDER BY escape_count DESC;
```

Comment:
According to the project requirements, please write the correct query for the question 👇 

![image](https://github.com/sergio-usma/vet_clinic/assets/78251927/e5eb25d5-4911-4e3a-85c6-fd7e8d3181fb)

Without the condition `escape_attempts > 0`🤓

- [ ] **I'm unable to add an inline comment in your readme file, please check the suggested change below**

![image](https://github.com/sergio-usma/vet_clinic/assets/78251927/ace9eb86-54f0-4ff1-ae3f-28d9d05b8f6d)

- [ ] Kindly add a license to your project to fix the broken link.

- [ ] Public repositories on GitHub are often used to share open-source software. For your repository to truly be open source, you'll need to license it so that others are free to use, change, and distribute the software.